src/components/astra-editor/plugins/prompt-dialog.ts

```diff
@@ -0,0 +1,318 @@
+import { css, html, LitElement, nothing, type PropertyValues } from 'lit'
+import { customElement, property, state } from 'lit/decorators.js'
+import { createRef, ref, type Ref } from 'lit/directives/ref.js'
+import { unsafeHTML } from 'lit/directives/unsafe-html.js'
+
+const MINIMAL_HEIGHT = 40
+
+const LoadingSvg = `<svg class='spin' xmlns="http://www.w3.org/2000/svg" width="13" height="13" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-loader">
+  <path d="M12 2v4"/>
+  <path d="m16.2 7.8 2.9-2.9"/>
+  <path d="M18 12h4"/>
+  <path d="m16.2 16.2 2.9 2.9"/>
+  <path d="M12 18v4"/>
+  <path d="m4.9 19.1 2.9-2.9"/><path d="M2 12h4"/>
+  <path d="m4.9 4.9 2.9 2.9"/>
+</svg>`
+
+@customElement('astra-editor-prompt-dialog')
+export default class PromptDialog extends LitElement {
```

Comment:
Just an FYI: I'm not suggesting you redo this, and it's kind of better the way it is, but you can extends `ClassifiedElement` and use tailwindcss classes